File: app/src/main/java/cn/intret/app/picgo/model/event/RescanImageDirectoryMessage.java
package cn.intret.app.picgo.model.event;

import java.io.File;

/**
 * 文件夹已经被重新扫描的消息
 */

public class RescanImageDirectoryMessage extends BaseMessage {
    private File mDirectory;

    public RescanImageDirectoryMessage() {
    }

    public RescanImageDirectoryMessage(File directory) {
        mDirectory = directory;
    }

    public RescanImageDirectoryMessage(String desc, File directory) {
        super(desc);
        mDirectory = directory;
    }

    public RescanImageDirectoryMessage setDirectory(File directory) {
        mDirectory = directory;
        return this;
    }

    public File getDirectory() {
        return mDirectory;
    }

    @Override
    public String toString() {
        return "RescanImageDirectoryMessage{" +
                "mDirectory=" + mDirectory +
                "} " + super.toString();
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/event/RescanImageDirectoryMessage.kt
package cn.intret.app.picgo.model.event

import java.io.File

/**
 * 文件夹已经被重新扫描的消息
 */

class RescanImageDirectoryMessage : BaseMessage {
    private var mDirectory: File? = null

    constructor()

    constructor(directory: File) {
        mDirectory = directory
    }

    constructor(desc: String, directory: File) : super(desc) {
        mDirectory = directory
    }

    fun setDirectory(directory: File): RescanImageDirectoryMessage {
        mDirectory = directory
        return this
    }

    fun getDirectory(): File? {
        return mDirectory
    }

    override fun toString(): String {
        return "RescanImageDirectoryMessage{" +
                "mDirectory=" + mDirectory +
                "} " + super.toString()
    }
}
